### ISSUE TITLE
Consumer group with random prefix

### ISSUE THREAD

**Issue by damccorm** [apache/beam#21259]
I'm running two beam pipelines on Apache Flink which read messages from the same Kafka topic by using KafkaIO. Both of them are in the same consumer group, so in my understanding each message should be consumed by either pipeline (but not both) but I observed that the same set of messages is consumed by each pipeline, meaning each message is processed at least twice. 
I can see in the log that additional consumer group with prefix "Reader-1_offset_consumer_<random_number\>_" was created and also consume messages. 
Could you explain what should I do to consume messages only once by one consumer group and why that additional group was created?

Similar issue: BEAM-12880

Imported from Jira [BEAM-12995](https://issues.apache.org/jira/browse/BEAM-12995). Original Jira may contain additional context.
Reported by: htsi.

**Comment by manuzhang** [comment]
Duplicate of https://github.com/apache/beam/issues/20159 per original discussion.
